theano/tensor/nnet/abstract_conv.py

```diff
@@ -137,6 +138,374 @@ def get_conv_shape_1axis(image_shape, kernel_shape, border_mode,
+        possibly depth) axis.
+    filter_dilation: tuple of int (symbolic or numeric). Its two or three
+        elements correspond respectively to the dilation on height and
+        width axis.
+
+    Returns
+    -------
+    Returns False if a convolution with the given input shape, kernel shape
+    and parameters would not have produced the given output shape.
+
+    Returns True in all other cases: if the given output shape matches the
+    computed output shape, but also if the shape could not be checked because
+    because the shape contains symbolic values.
+
+    """
+    image_shape = tuple(image_shape)
+    kernel_shape = tuple(kernel_shape)
+    output_shape = tuple(output_shape)
+
+    if len(image_shape) != len(kernel_shape) or len(image_shape) != len(output_shape):
+        return False
+    if len(image_shape) - 2 != len(subsample):
+        return False
+    if filter_dilation is not None and len(image_shape) - 2 != len(filter_dilation):
+        return False
+
+    # compute the predicted output shape
+    computed_output_shape = get_conv_output_shape(
+        image_shape, kernel_shape, border_mode, subsample, filter_dilation)
+
+    # check if the given output shape matches the computed shape
+    def check_dim(given, computed):
+        if given is None or computed is None:
+            return True
+        try:
+            given = get_scalar_constant_value(given)
+            computed = get_scalar_constant_value(computed)
+            return int(given) == int(computed)
+        except NotScalarConstantError:
+            # no answer possible, accept for now
+            return True
+
+    return all(check_dim(given, computed)
+               for (given, computed) in zip(output_shape, computed_output_shape))
+
+
+def assert_conv_shape(shape):
+    """This function adds Assert nodes that check if shape is a valid convolution shape.
+
+    The first two dimensions should be larger than or equal to zero. The convolution
+    dimensions should be larger than zero.
+
+    Parameters
+    ----------
+    shape: tuple of int (symbolic or numeric) corresponding to the input, output or
+        kernel shape of a convolution. For input and output, the first elements should
+        should be the batch size and number of channels. For kernels, the first and
+        second elements should contain the number of input and output channels.
+        The remaining dimensions are the convolution dimensions.
+
+    Returns
+    -------
+    Returns a tuple similar to the given `shape`. For constant elements in `shape`,
+    the function checks the value and raises a `ValueError` if the dimension is invalid.
+    The elements that are not constant are wrapped in an `Assert` op that checks the
+    dimension at run time.
+    """
+    out_shape = []
+    for i, n in enumerate(shape):
+        try:
+            const_n = get_scalar_constant_value(n)
+            if i < 2:
```

Comment:
I think 2 is correct, actually. This is a test for the first two dimensions, for the batch size and number of channels. Those can be equal to zero, the convolution dimensions (i >= 2) not.